aggregator_api/src/routes.rs

```diff
@@ -52,18 +52,24 @@ pub(super) async fn get_config(
     })
 }
 
+pub(crate) struct LowerBound(Option<TaskId>);
+#[async_trait]
+impl TryFromConn for LowerBound {
+    type Error = Error;
+    async fn try_from_conn(conn: &mut Conn) -> Result<Self, Self::Error> {
+        querify(conn.querystring())
+            .into_iter()
+            .find(|&(k, _)| k == "pagination_token")
+            .map(|(_, v)| TaskId::from_str(v))
+            .transpose()
+            .map(Self)
+            .map_err(|err| Error::BadRequest(format!("Couldn't parse pagination_token: {:?}", err)))
+    }
+}
+
 pub(super) async fn get_task_ids<C: Clock>(
-    conn: &mut Conn,
-    State(ds): State<Arc<Datastore<C>>>,
+    (LowerBound(lower_bound), ds): (LowerBound, Store<C>),
```

Comment:
as this is the first instance of this: I replaced a lot of restructuring of State and Json anywhere the contents don't need to be moved out. It's less verbose to dereference through the newtype